nemo/core/classes/modelPT.py

```diff
@@ -71,27 +73,34 @@ def setup_optimization(self, optim_params: Optional[Dict] = None) -> Optimizer:
         Returns:
             An instance of a torch.optim.Optimizer
         """
-        optim_params = optim_params or {}  # In case null was passed as optim_params
-
-        # Check if caller provided optimizer name, default to Adam otherwise
-        optimizer_name = optim_params.get('optimizer', 'adam')
-
-        # Check if caller has optimizer kwargs, default to empty dictionary
-        optimizer_args = optim_params.get('opt_args', [])
-        optimizer_args = parse_optimizer_args(optimizer_args)
-
-        # We are guarenteed to have lr since it is required by the argparser
-        # But maybe user forgot to pass it to this function
-        lr = optim_params.get('lr', None)
-
-        if 'lr' is None:
-            raise ValueError('`lr` must be passed to `optim_params` when setting up the optimization !')
-
-        # Actually instantiate the optimizer
-        optimizer = get_optimizer(optimizer_name)
-        optimizer = optimizer(self.parameters(), lr=lr, **optimizer_args)
-
-        # TODO: Remove after demonstration
-        logging.info("Optimizer config = %s", str(optimizer))
+        if isinstance(optim_params, Dict):
+            optim_params = optim_params or {}  # In case null was passed as optim_params
+
+            # Check if caller provided optimizer name, default to Adam otherwise
+            optimizer_name = optim_params.get('optimizer', 'adam')
+
+            # Check if caller has optimizer kwargs, default to empty dictionary
+            optimizer_args = optim_params.get('opt_args', [])
+            optimizer_args = parse_optimizer_args(optimizer_args)
+
+            # We are guarenteed to have lr since it is required by the argparser
+            # But maybe user forgot to pass it to this function
+            lr = optim_params.get('lr', None)
+
+            if 'lr' is None:
+                raise ValueError('`lr` must be passed to `optimizer_config` when setting up the optimization !')
+
+            # Actually instantiate the optimizer
+            optimizer = get_optimizer(optimizer_name)
+            optimizer = optimizer(self.parameters(), lr=lr, **optimizer_args)
+
+            # TODO: Remove after demonstration
+            logging.info("Optimizer config = %s", str(optimizer))
```

Comment:
Remove?

Reply:
Or will @titu1994 clean this part up?

Reply:
I'm intending to keep it - it provides a good overview of default parameters for the optimizer and whether the overrides worked or not.

Reply:
I removed it.